### PR TITLE
Add "Mirror" mod to osu!catch

### PR DESCRIPTION
**Pull request by ggliv**
Previously discussed in #7334.

Flips catch objects horizontally.

### CODE REVIEW

**Review by frenzibyte — DISMISSED** [review]

**Comment by peppy** [comment]
This looks incorrect for nested droplets:

20210809 160517 (dotnet)
20210809 160544 (dotnet)

Also this should be applied to banana showers, I think.

**Review by peppy — DISMISSED** [review]
As proposed

**Comment by bdach** [comment]
I believe I've fixed the issues mentioned above.

| no mirror | mirror |
| :-: | :-: |
| osu_2021-08-10_22-35-26 | osu_2021-08-10_22-36-25 |
| osu_2021-08-10_22-35-51 | osu_2021-08-10_22-36-43 |

Test case that aims to demonstrate this also attached.

**Review by frenzibyte — APPROVED** [review]

File: osu.Game.Rulesets.Catch/Mods/CatchModMirror.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        /// </summary>
+        private static void mirrorEffectiveX(CatchHitObject catchObject)
+        {
+            catchObject.OriginalX = CatchPlayfield.WIDTH - catchObject.OriginalX;
+            catchObject.XOffset = -catchObject.XOffset;
+
+            foreach (var nested in catchObject.NestedHitObjects.Cast<CatchHitObject>())
+            {
+                nested.OriginalX = CatchPlayfield.WIDTH - nested.OriginalX;
+                nested.XOffset = -nested.XOffset;
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Catch/Mods/CatchModMirror.cs:62]: nit: Could've been a recursive invocation.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**